LFS-925: Update questionnaires for cards4care

acrowthe opened the pull request:
Based on #483 to avoid merge conflicts

acrowthe committed:
LFS-925: Update questionnaires for cards4care
Implement changes

acrowthe committed:
LFS-925: Update questionnaires for cards4care
Fix Graded Walk Test and Daily Actigraph Diary issues

k-pace commented:
Found issue with cpet_end_criteria options - fixed on spreadsheet, line breaks for separation were not properly done by me in the spreadsheet

sashaandjic commented:
**In CPET Interpretation this item is not done:**
Criteria to formally end protocol – separate chest pain/tightness from Difficulty tolerating mouthpiece/nose clip, separate External interruption from Maximum heart rate, patient requested to stop and ST changes. These should all be separate options
![CPET interpretation](https://user-images.githubusercontent.com/23243244/108470011-45944c00-7257-11eb-9744-a509456d4fd6.png)

The rest is good
-------------------------------------------------------------------------------------------------------------------
Graded Walk Test
![image](https://user-images.githubusercontent.com/23243244/108466899-60b08d00-7252-11eb-9778-b2dcca60bd64.png)

Daily Actigraph Diary
![Daily Actigraph Diary](https://user-images.githubusercontent.com/23243244/108467581-6ce91a00-7253-11eb-9770-49ef147f07e1.png)

Event: Clinical - free text Diagnosis option added; Stress test results no longer required (perhaps we can make this field conditional? - if clinicalTestsPerformed)
![Event Clinical](https://user-images.githubusercontent.com/23243244/108468286-8179e200-7254-11eb-9f51-ce548761bdf4.png)

Event Medication Change: Should Medication Category should stay as single value?
![Event Medication Change](https://user-images.githubusercontent.com/23243244/108469335-0ca7a780-7256-11eb-8326-a96fce0eaf22.png)

Baseline Health Information
![Baseline Health Information](https://user-images.githubusercontent.com/23243244/108470248-ac196a00-7257-11eb-8911-d6cab6d9b775.png)

sashaandjic commented:
> * In 'Event: Medication Change', selecting Other Medication does not trigger a free text box. We can also just add this as a line in the set of medications like in other questions, rather than making it a separate question. The issue was probably that the spreadsheet has 'Y if previous = other' - I corrected this now in the spreadsheet

It is a conditional field `Specify other medication(s)`, not the free text in The Baseline Medication form.

sashaandjic commented:
> * I did a poor job of explaining the sections for Daily Actigraph Diary - the entirety of the current questions make up a day, and this day needs to be repeatable (time put on through 'was physical activity performed' and all it's sub questions. We should have an 'add entry' button at the bottom of this question set to create an entry for the next date

Isn't it better to leave it as is? Right now each day can be recorded in a separate form. This way, each day will be listed separately on the Patient chart. If we keep appending dates to the form, it will be very difficult to read and edit the form after 2 days of data.

k-pace commented:
> > * I did a poor job of explaining the sections for Daily Actigraph Diary - the entirety of the current questions make up a day, and this day needs to be repeatable (time put on through 'was physical activity performed' and all it's sub questions. We should have an 'add entry' button at the bottom of this question set to create an entry for the next date
> 
> Isn't it better to leave it as is? Right now each day can be recorded in a separate form. This way, each day will be listed separately on the Patient chart. If we keep appending dates to the form, it will be very difficult to read and edit the form after few days of data being added.
> This form is already quite heavy with 2 repeatable sections: ACTIGRAPH BREAK and ACTIVITY#. Wrapping the whole thing in another repeatable section will be a nightmare for users.

@sashaandjic OK - Agreed, let's do that instead; single day per form as it is right now!

k-pace commented:
There is one other modification that needs to be made if someone can pick it up (@marta- or @sdumitriu ?)
Graded walk test - Comments -> should not be required, removed minimum answers from xml

marta- commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Graded Walk Test.xml:
```diff
@@ -0,0 +1,283 @@
+				<value>text</value>
+				<type>String</type>
+			</property>
+			<property>
+				<name>minAnswers</name>
+				<value>1</value>
+				<type>Long</type>
+			</property>
```
```suggestion
```

sashaandjic commented:
small issue found - no value check on Intensity field
![image](https://user-images.githubusercontent.com/23243244/108572963-791bb880-72e1-11eb-9205-9d54a9d720d0.png)

sashaandjic commented:
The highlighted texts should be separate options. Please change "**difficult** tolerating mouthpiece/nose clip" to "**Difficulty** tolerating..."
![image](https://user-images.githubusercontent.com/23243244/108573925-e16b9980-72e3-11eb-86ec-49f77f0970df.png)

sashaandjic commented:
Event: Medication Change - either replace Other medication option with free text, or add conditional field` Specify other medication` when Other medication is selected
![image](https://user-images.githubusercontent.com/23243244/108574149-8f774380-72e4-11eb-9c9a-cfdefeb58179.png)

sashaandjic commented:
On Graded walk test form `Comments `filed should not be required

acrowthe committed:
LFS-925: Update questionnaires for cards4care
- Graded walk test: comment no longer required
- Event - Medication Change: `Other medication` option renamed to `Other`, is now a free text field. Following `Explain` field is no longer needed as a result, condition changed to `other` so it will be skipped by the importer.
- Event - Clinical: Added condition `N of ce_test = 1` to `Stress test results`. Still optional, but now hidden unless a test was performed.
- Daily Actigraph Diary
  - `Was physical activity performed?` is now required
  - minValue 6 and maxValue 20 added to `Intensity` question
- CPET Interpretation: options fixed for `What criteria was formally used to end the protocol?`

acrowthe commented:
Changes:
- Graded walk test: comment no longer required
- Event - Medication Change: `Other medication` option renamed to `Other`, is now a free text field. Following `Explain` field is no longer needed as a result, condition changed to `other` so it will be skipped by the importer.
- Event - Clinical: Added condition `N of ce_test = 1` to `Stress test results`. Still optional, but now hidden unless a test was performed.
- Daily Actigraph Diary
  - `Was physical activity performed?` is now required
  - minValue 6 and maxValue 20 added to `Intensity` question
- CPET Interpretation: options fixed for `What criteria was formally used to end the protocol?`

sashaandjic reviewed (approved):
tested

k-pace commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Graded Walk Test.xml:
```diff
@@ -0,0 +1,278 @@
+			<primaryNodeType>lfs:Question</primaryNodeType>
+			<property>
+				<name>text</name>
+				<value>Graded walk test number</value>
```
Change to Graded walk test visit number

k-pace commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Graded Walk Test.xml:
```diff
@@ -0,0 +1,278 @@
+			</property>
+			<property>
+				<name>maxAnswers</name>
+				<value>1</value>
```
change to '3'

k-pace commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Graded Walk Test.xml:
```diff
@@ -0,0 +1,278 @@
+				<primaryNodeType>lfs:AnswerOption</primaryNodeType>
+				<property>
+					<name>label</name>
+					<value>1</value>
```
change to '3'

k-pace commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Graded Walk Test.xml:
```diff
@@ -0,0 +1,278 @@
+				<primaryNodeType>lfs:AnswerOption</primaryNodeType>
+				<property>
+					<name>label</name>
+					<value>2</value>
```
change to 5

k-pace commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Graded Walk Test.xml:
```diff
@@ -0,0 +1,278 @@
+				<primaryNodeType>lfs:AnswerOption</primaryNodeType>
+				<property>
+					<name>label</name>
+					<value>3</value>
```
6

k-pace commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Graded Walk Test.xml:
```diff
@@ -0,0 +1,278 @@
+				<primaryNodeType>lfs:AnswerOption</primaryNodeType>
+				<property>
+					<name>label</name>
+					<value>4</value>
```
8

acrowthe committed:
LFS-925: Update questionnaires for cards4care
Fix Graded Walk Test number

sashaandjic reviewed (changes requested):
Visit number not added to the CPET Interpretation form

acrowthe committed:
LFS-925: Update questionnaires for cards4care
Add CPET visit number

sashaandjic commented:
Why CPET Interpretation and Graded Walk Test have visit# as radio buttons, while 6MWT, Baseline Cardiovascular Parameters, Baseline Medications and Surveys have simple text field?
![image](https://user-images.githubusercontent.com/23243244/108759290-2a0c9800-751a-11eb-882b-f297e938c56a.png)
![image](https://user-images.githubusercontent.com/23243244/108759223-15300480-751a-11eb-89ab-a7cab1b82687.png)

acrowthe committed:
LFS-925: Update questionnaires for cards4care
Update all visit number questions to radio buttons

acrowthe committed:
LFS-925: Update questionnaires for cards4care
Update Actigraph Data maxPerSubject property and visit number question

sashaandjic reviewed (approved):
tested

sdumitriu commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Baseline Medications.xml:
```diff
@@ -369,7 +430,7 @@
 			<property>
 				<name>description</name>
-				<value>excluding beta-blockers and CCBs</value>
+				<value>Excluding beta-blockers and CCBs</value>
```
I made this lowercase to match all the other descriptions.

sdumitriu commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Event - Medication Change.xml:
```diff
@@ -242,30 +213,30 @@
 		</property>
 		<node>
-			<name>Antiarrhythmic </name>
+			<name>Antiarrhythmic (eg Amiodarone, Sotalol, Mexiletene)</name>
```
I prefer not putting special characters in node names.

sdumitriu commented on cardiac-rehab-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Surveys.xml:
```diff
@@ -1604,6 +1657,11 @@
 					<type>String</type>
 				</property>
+				<property>
+					<name>notApplicable</name>
```
I'm not sure marking these as `notApplicable` is correct. `None` is a perfectly valid answer to a question like "What is your level of difficulty in running a short distance?"
